File: core/src/commands/process.rs
use sysinfo::{Pid, ProcessExt, SystemExt};

use crate::{
    core::injection::{get_roblox_processes, Process},
    util::consts::SYS_INFO,
};

#[tauri::command]
pub async fn kill_process(pid: usize) -> Result<(), String> {
    // kill the process

    SYS_INFO
        .lock()
        .await
        .process(Pid::from(pid))
        .unwrap()
        .kill();

    Ok(())
}

#[tauri::command]
pub async fn get_processes() -> Result<Vec<Process>, String> {
    let processes = get_roblox_processes();

    Ok(processes)
}

File: src/lib/util/process.ts
import { Command } from "@tauri-apps/api/shell";
import { invoke } from "@tauri-apps/api/tauri";

export interface GetProcessesResult {
    pid?: number;
    command?: string;
    arguments: string[];
}

export function killProcess(pid: number) {
    new Command("kill", ["-9", pid.toString()]).spawn();
}

export async function getRobloxProcesses(): Promise<GetProcessesResult[]> {
    return invoke("get_processes");
}
